block/src/Pos/Block/Network/Retrieval.hs

```diff
@@ -308,3 +309,50 @@ getProcessBlocks pm diffusion nodeId desired checkpoints = do
                   else pure False
           when exitedRecovery $
               logInfo "Recovery mode exited gracefully on receiving block we needed"
+
+streamProcessBlocks
+    :: forall ctx m.
+       ( BlockWorkMode ctx m
+       , HasMisbehaviorMetrics ctx
+       )
+    => ProtocolMagic
+    -> Diffusion m
+    -> NodeId
+    -> HeaderHash
+    -> [HeaderHash]
+    -> m ()
+streamProcessBlocks pm diffusion nodeId desired checkpoints = do
+    logInfo "streaming start"
+    r <- Diffusion.streamBlocks diffusion nodeId desired checkpoints (loop (0::Word32) [])
+    case r of
+         Nothing -> do
+             logInfo "streaming not supported, reverting to batch mode"
+             getProcessBlocks pm diffusion nodeId desired checkpoints
+         Just _  -> do
+             logInfo "streaming done"
+             return ()
+  where
+    loop n blocks (streamWindow, wqgM, blockChan) = do
+        streamEntry <- atomically $ readTBQueue blockChan
+        case streamEntry of
+          StreamEnd         -> addBlocks blocks
+          StreamBlock block -> do
+              let batchSize = min 64 streamWindow
+              let !n' = n + 1
+              when (n' `mod` 256 == 0) $
+                     logInfo $ sformat ("Read block "%shortHashF%" difficulty "%int) (headerHash block)
+                                        (block ^. difficultyL)
```

Comment:
Why do this? Is it just for debugging?

Reply:
Only printing it for every 256:th block limits log flooding.

Reply:
Ok I'm not strictly opposed to it, but it should be debug-level.

Reply:
Will change it to debug.